File: APIs/mixitup-typescript-api/lib/commands.ts
import { RestClient } from "./restclient";
import { ICommand } from "./models/command";

export class Commands {
    public static async getAllCommandsAsync(): Promise<ICommand[]> {
        return await RestClient.GetAsync<ICommand[]>("commands");
    }

    public static async getCommandAsync(commandID: string): Promise<ICommand> {
        return await RestClient.GetAsync<ICommand>(`commands/${commandID}`);
    }

    public static async runCommandAsync(commandID: string): Promise<{}> {
        return await RestClient.PostAsync<{}>(`commands/${commandID}`, null);
    }

    public static async updateCommandAsync(updatedCommand: ICommand): Promise<ICommand> {
        return await RestClient.PutAsync<ICommand>(`commands/${updatedCommand.ID}`, updatedCommand);
    }
}

File: APIs/MixItUp.API/Commands.cs
using MixItUp.API.Models;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace MixItUp.API
{
    //  GET:    http://localhost:8911/api/commands              Commands.GetAllCommandsAsync
    //  GET:    http://localhost:8911/api/commands/{commandID}  Commands.GetCommandAsync
    //  POST:   http://localhost:8911/api/commands/{commandID}  Commands.RunCommandAsync
    //  PUT:    http://localhost:8911/api/commands/{commandID}  Commands.UpdateCommandAsync
    public static class Commands
    {
        public static async Task<Command[]> GetAllCommandsAsync()
        {
            return await RestClient.GetAsync<Command[]>($"commands");
        }

        public static async Task<Command> GetCommandAsync(Guid commandID)
        {
            return await RestClient.GetAsync<Command>($"commands/{commandID}");
        }

        public static async Task RunCommandAsync(Guid commandID)
        {
            await RunCommandAsync(commandID, null);
        }

        public static async Task RunCommandAsync(Guid commandID, IEnumerable<string> arguments)
        {
            await RestClient.PostAsync($"commands/{commandID}", arguments);
        }

        public static async Task<Command> UpdateCommandAsync(Command updatedCommand)
        {
            return await RestClient.PutAsync<Command>($"commands/{updatedCommand.ID}", updatedCommand);
        }
    }
}
